src/oncall/scheduler/round-robin.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime, timedelta
+from pytz import timezone, utc
+from falcon import HTTPBadRequest
+import time
+import default
+import logging
+
+logger = logging.getLogger()
+
+class Scheduler(default.Scheduler):
+
+    def guess_last_scheduled_user(self, schedule, start, roster, cursor):
+        cursor.execute('''
+                        SELECT `user_id`, MAX(`end`) AS `last_end` FROM `event`
+                        WHERE `team_id` = %s AND `user_id` IN %s AND `end` <= %s
+                        AND `role_id` = %s
+                        GROUP BY `user_id`
+                        ''', (schedule['team_id'], roster, start, schedule['role_id']))
+        if cursor.rowcount != 0:
+            # Grab user id with largest last scheduled time
+            return max(cursor.fetchall(), key=lambda x: x['last_end'])['user_id']
+        else:
+            return None
+
+    def find_least_active_available_user_id(self, schedule, future_events, cursor):
+        # Ordered by roster priority, break potential ties with user id
+        cursor.execute('SELECT * FROM roster_user WHERE roster_id = %s AND in_rotation = 1 ORDER BY roster_priority, user_id',
+                       schedule['roster_id'])
+        roster = [row['user_id'] for row in cursor]
+        cursor.execute('SELECT last_scheduled_user_id FROM schedule WHERE id = %s', schedule['id'])
+        if cursor.rowcount == 0 or roster == []:
+            # Schedule doesn't exist, or roster is empty. Bail
+            return None
+        last_user = cursor.fetchone()['last_scheduled_user_id']
+        if last_user not in roster:
+            # If this user is no longer in the roster or last_scheduled_user in NULL, try to find
+            # the last scheduled user using the calendar
+            start = min([e['start'] for e in future_events])
```

Comment:
you can make this a generator expression by removing the surrounding [] from the list comp